Dockerfile

```diff
@@ -1,6 +1,8 @@
-FROM golang:latest AS BUILDER
+FROM asecurityteam/sdcli:v1 AS BUILDER
+RUN mkdir -p /go/src/github.com/asecurityteam/transportd
 WORKDIR /go/src/github.com/asecurityteam/transportd
-COPY . .
+COPY --chown=sdcli:sdcli . .
+RUN sdcli go dep
```

Comment:
how were dependencies brought in before this line?

Reply:
I believe we have to keep both as they cover different cases. During the Travis run we pull in the dependencies and run tests but we don't use the `Dockerfile` during any of the testing stages. Currently, we only use the `Dockerfile` to generate the published image.

When we first made the `Dockerfile` we assumed it would be used during the final stage of a Travis job to build and push the image to Docker Hub. However, Docker Hub builds actually happen on the Docker Hub infrastructure and are detached from our CI/CD pipelines (triggered on merge). The result is that the Docker Hub builds fail because the freshly cloned repo doesn't contain any of the dependencies.

My preference would be to build and push the image from Travis so that it's guaranteed to have the same dependency set as we tested with rather than pulling them down again during the image build. I put a ticket in our backlog to correct this. In the meantime, this seemed like the most expedient way to get the builds running.